Raspberry Pi/RaspberrySide/VisionManager.py

```diff
@@ -83,18 +84,28 @@ def updateImage(self):
         if self.isObjectDetected:
             self.currentImageObject.didUpdateVar = False	
 
-        didGetImage,frame = self.cam.read()
+        didGetImage, new_frame = self.cam.read()
+        if not didGetImage:  # Can't event grab one frame - might be a problem
+            logger.error("Couldn't Read Image from self.cam!")
+            return
+        
+        for _ in xrange(self.READ_BUFFER_AMOUNT - 1):  # minus one, since we already grabbed one frame.
```

Comment:
Another suggestion was to run the `cam.read()` several times sequentially - in order to clear any buffer.
This needs to account for the difference in capture rates: i.e: if the camera works at 24 fps, but we only read 4 fps we need to read 6 frames, throw away 5 and only parse the last one.

The code can still be kept while setting `READ_BUFFER_AMOUNT = 0` in order to disable the buffer read.